### PR TITLE
Store `mlflow.runName` tag when creating or updating run

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Store `mlflow.runName` tag when creating or updating run.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/tracking/file_store.py [hunk 1]
```diff
@@ -615,6 +617,9 @@ def create_run(self, experiment_id, user_id, start_time, tags, run_name):
         mkdir(run_dir, FileStore.METRICS_FOLDER_NAME)
         mkdir(run_dir, FileStore.PARAMS_FOLDER_NAME)
         mkdir(run_dir, FileStore.ARTIFACTS_FOLDER_NAME)
+        tags = tags or []
+        if run_name is not None:
```
Comment (BenWilson2) [inline mlflow/store/tracking/file_store.py:621]: Is it ever not going to be `None` any longer?

Comment (harupy) [reply mlflow/store/tracking/file_store.py:621]: On line 596, we run `run_name = run_name if run_name is not None else _generate_random_name()`. `run_name` never becomes None.

Comment (harupy) [reply mlflow/store/tracking/file_store.py:621]:
```suggestion
```

File: mlflow/store/tracking/sqlalchemy_store.py [hunk 2]
```diff
@@ -547,8 +548,10 @@ def create_run(self, experiment_id, user_id, start_time, tags, run_name):
                 lifecycle_stage=LifecycleStage.ACTIVE,
             )
 
-            if tags is not None:
-                run.tags = [SqlTag(key=tag.key, value=tag.value) for tag in tags]
+            tags = tags or []
+            if run_name is not None:
```
Comment (harupy) [inline mlflow/store/tracking/sqlalchemy_store.py:552]:
```suggestion
```

**Review by BenWilson2 — APPROVED** [review]
LGTM!

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

**Pull request merged**